### DISCUSSION TITLE
Gem::Deprecate error on macOS

### DISCUSSION THREAD

**Discussion by jamesandies** [Homebrew/discussions#3354] [Getting started] [answered]
I think I may have had my laptop die during a "brew update" and its incompletion has left me with errors. For any brew commands, I get the error:

/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/rubygems.rb:596:in `singleton class': uninitialized constant Gem::Deprecate (NameError)
	from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/rubygems.rb:595:in `<module:Gem>'
	from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/rubygems.rb:116:in `<top (required)>'
	from <internal:gem_prelude>:2:in `require'
	from <internal:gem_prelude>:2:in `<internal:gem_prelude>'
Failed during: /usr/local/bin/brew update --force --quiet

What I have tried is,
- installing & uninstalling the "sorbet-runtime" gem (as per `https://github.com/Homebrew/brew/issues/8908` and everything linked to it)
- uninstalling all gems
- rerunning the brew install script
- uninstalling brew and re-installing it

None of these have worked. I'm on macOS 10.15.7 and I have no idea how to fix this now. Please help :'(

**Comment by gromgit** [comment, marked as answer]
> * uninstalling brew and re-installing it

That should absolutely have fixed your problem, assuming everything else had failed. The uninstaller would've told you about certain files and directories that it could not clean up for some reason; did you manually clean those up before running the installer again?

**Reply by Bo98** [reply]
Even just reinstalling the Portable Ruby part should work: `rm -r /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby && brew update`

**Reply by jamesandies** [reply]
i ran the uninstall using,
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/uninstall.sh)"

and this is the response I got:

```
Warning: This script will remove:
/Users/me/Library/Caches/Homebrew/
/usr/local/Caskroom/
/usr/local/Cellar/
/usr/local/bin/brew -> /usr/local/bin/brew
Are you sure you want to uninstall Homebrew? This will remove your installed packages! [y/N] y
==> Removing Homebrew installation...
Warning: Failed to delete /usr/local/Caskroom
rm: /usr/local/Caskroom: Permission denied
Warning: Failed to delete /usr/local/Cellar
rm: /usr/local/Cellar: Permission denied
==> Removing empty directories...
Password:
==> /usr/bin/sudo /usr/bin/find /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/opt /usr/local/sbin /usr/local/share /usr/local/var /usr/local/Caskroom /usr/local/Cellar /usr/local/Homebrew /usr/local/Frameworks -name .DS_Store -delete
==> /usr/bin/sudo /usr/bin/find /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/opt /usr/local/sbin /usr/local/share /usr/local/var /usr/local/Caskroom /usr/local/Cellar /usr/local/Homebrew /usr/local/Frameworks -depth -type d -empty -exec rmdir {} ;
==> Homebrew uninstalled!
The following possible Homebrew files were not deleted:
/usr/local/.com.apple.installer.keep
/usr/local/Frameworks/
/usr/local/Homebrew/
/usr/local/aria2/
/usr/local/bin/
/usr/local/etc/
/usr/local/include/
/usr/local/lib/
/usr/local/opt/
/usr/local/sbin/
/usr/local/share/
/usr/local/var/
You may wish to remove them yourself.
```

Should I remove all of the remaining directories?

**Reply by gromgit** [reply]
Since the errors you got come from files under `/usr/local/Homebrew/`, and that's clearly a Homebrew-specific directory, you really should nuke that before reinstalling. `/usr/local/aria2/` is also specific to `aria2`, which I assume you installed via Homebrew, so that can go too.

**Reply by jamesandies** [reply]
Cleared out /usr/local/Homebrew and re-installed brew and it seems like everything is back to normal. Thank you
